netlogo-core/src/main/api/XMLUtils.scala

```diff
@@ -0,0 +1,136 @@
+// (C) Uri Wilensky. https://github.com/NetLogo/NetLogo
+
+package org.nlogo.core
+
+import java.io.{ StringReader, Writer }
+import javax.xml.stream.{ XMLInputFactory, XMLOutputFactory, XMLStreamConstants, XMLStreamException }
+
+import scala.util.{ Failure, Try }
+import scala.util.matching.Regex
+
+object XMLUtils {
```

Comment:
A few small things here....

The file is `XMLUtils.scala`, and then it has `object XMLUtils` and `class XMLWriter`.  All of those terms should probably be the same.

If you have the `CDATA_ESCAPE` thing in the companion object for the class (i.e. `object XMLWriter` and `class XMLWriter`), you get the added benefit of not needing to prefix it as `XMLUtils.CDATA_ESCAPE` (so you can just reference it as `CDATA_ESCAPE`).  A `val` inside of a companion object is equivalent to having a `static` member on a class in Java.

Lastly (and very minorly), all-caps is unidiomatic for Scala constants.  Go with `CDataEscape` or `CDATAEscape` instead.

Reply:
If you didn't already see this, XMLUtils.scala contains both the XMLWriter and XMLReader class, both of which access the escape variable, so I didn't want it to only live in an object definition for XMLWriter. If you have a suggestion for a better name, I'm open to changing it.